src/libmongoc/src/mongoc/mongoc-scram.c

```diff
@@ -1150,4 +1229,176 @@ _mongoc_sasl_prep (const char *in_utf8, int in_utf8_len, bson_error_t *err)
    return bson_strdup (in_utf8);
 #endif
 }
+
+int
+_mongoc_utf8_char_length (const char *c)
+{
```

Comment:
```suggestion
{
   BSON_ASSERT_PARAM (c);

```
Assert parameters that must not be NULL.